pkg/eventqueue/eventqueue.go

```diff
@@ -309,9 +316,19 @@ func (q *EventQueue) WaitToBeDrained() {
 	}
 	<-q.close
 
-	// In-flight events may still be running. Wait for them to be completed for
-	// the queue to be fully drained.
-	<-q.eventsClosed
+	// If the queue is running, then in-flight events may still be ongoing.
+	// Wait for them to be completed for the queue to be fully drained. If the
+	// queue is not running, we must forcefully drain it because nothing else
+	// will.
+	if running := q.isRunning.Load().(bool); running {
+		<-q.eventsClosed
+	} else {
+		for ev := range q.events {
+			// Close the results channel on the event to stop receivers from
+			// blocking.
+			close(ev.eventResults)
```

Comment:
This will have a different behavior, i.e., we are not "canceling" similarly what is being done on line 263.

Discussed offline.